crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -684,7 +737,11 @@ impl SpecializedMeshPipeline for MeshPipeline {
             depth_stencil: Some(DepthStencilState {
                 format: TextureFormat::Depth32Float,
                 depth_write_enabled,
-                depth_compare: CompareFunction::Greater,
+                depth_compare: if key.contains(MeshPipelineKey::PREPASS_DEPTH) {
+                    CompareFunction::Equal
+                } else {
+                    CompareFunction::GreaterEqual
+                },
```

Comment:
this codepath is never hit. the PREPASS_DEPTH flag is only set when rendering the prepass itself, which doesn't use the base mesh pipeline. If it WAS hit, then alpha-blend materials would not render (depth would not be equal), and others may flicker due to lack of @invariant flag. 

We do need to change this to GreaterEqual so that depth tests still pass with the depth buffer populated, but I don't think we need Equal in any circumstances.

Reply:
Right, I remember trying to use the mesh pipeline specialization to avoid copying some of the shader_defs creation, but I ended up not using it. I guess this was some left overs I didn't remove. Good catch!